app/gui/src/project-view/util/ast/match.ts

```diff
@@ -48,31 +62,32 @@ export class Pattern {
   }
 
   /** Create a new concrete example of the pattern, with the placeholders replaced with the given subtrees. */
-  instantiate(edit: Ast.MutableModule, subtrees: Ast.Owned[]): Ast.Owned {
+  instantiate(edit: Ast.MutableModule, subtrees: Ast.Owned<Ast.MutableExpression>[]): Ast.Owned<T> {
     const template = edit.copy(this.template)
     const placeholders = findPlaceholders(template, this.placeholder).map((ast) => edit.tryGet(ast))
     for (const [placeholder, replacement] of zipLongest(placeholders, subtrees)) {
       assertDefined(placeholder)
       assertDefined(replacement)
       placeholder.replace(replacement)
     }
-    return template
+    // Cast `Owned<Mutable<T>>` to `Owned<T>`. We require `T extends MutableAst`.
+    return template as any
```

Comment:
This cast could be a short function `dropMutability` with appropriate type assertions. Much harder to accidentally violate assumptions.

Reply:
I'll just return `Owned<Mutable<T>>`, it's the same as `Owned<T>` for all concrete `T extends MutableAst`.